cfme/cloud/provider/openstack.py

```diff
@@ -24,28 +27,21 @@ class OpenStackProvider(CloudProvider):
     _canvas_element = '//*[@id="noVNC_canvas"]'
     _ctrl_alt_del_xpath = '//*[@id="sendCtrlAltDelButton"]'
 
-    def __init__(self, name=None, endpoints=None, zone=None, key=None, hostname=None,
-                 ip_address=None, api_port=None, api_version=None, sec_protocol=None,
-                 amqp_sec_protocol=None, keystone_v3_domain_id=None, tenant_mapping=None,
-                 infra_provider=None, appliance=None):
-        super(OpenStackProvider, self).__init__(name=name, endpoints=endpoints,
-                                                zone=zone, key=key, appliance=appliance)
-        self.hostname = hostname
-        self.ip_address = ip_address
-        self.api_port = api_port
-        self.api_version = api_version
-        self.keystone_v3_domain_id = keystone_v3_domain_id
-        self.infra_provider = infra_provider
-        self.sec_protocol = sec_protocol
-        self.tenant_mapping = tenant_mapping
-        self.amqp_sec_protocol = amqp_sec_protocol
+    api_port = attr.ib(default=None)
```

Comment:
**Question/Required**

Looks like hostname gets dropped from the attribute list here

Reply:
this is ok because hostname and ip_address are defined in default endpoint. So, BaseProvider has appropriate setter/getter methods for them.